### PR TITLE
Rewrite the mount resource to fix at least one existing issue

### PR DESCRIPTION
**Pull request by ffrank**
Let's go. It's still not perfect, but simpler and more correct.

### CODE REVIEW

**Comment by ffrank** [comment]
Fixes #525

**Comment by ffrank** [comment]
The wrapping of resources is based on how NspawnRes does it.

But I wonder: It would feel much better to just run Watch of both the wrapped file and svc, and in the mount Watch, only wait for the svc to finish, and then also finish.

**Comment by purpleidea** [comment]
Hey great to see this! I had a quick look first. Lots of nice work there. Here are the open issues to discuss:

1. This removes fstab support-- I actually think that is still valuable to have, at least for now-- so how do we handle that? Do we have a separate resource? Do we have it be an optional for this resource? Do we have mount:systemd and mount:fstab ? Also see 2...

2. Is the systemd variant of this resource (as shown here) similar enough to the svc resource that they should just be combined there? I think architecturally they're quite similar. I would need to investigate more.

3. I'm not convinced I want shell tests to stay around. So we should consider a better approach if possible. This won't block your patch, but I want to mention it in case you have ideas.

Thanks Felix!

**Comment by ffrank** [comment]
1. It's a big hassle in terms of required code. I will say, what was there seemed to do a good enough job. The unit tests were nicely elaborate, BUT most of them never ran (they were written as methods of MountRes...). Then again, this feature opens us up to many additional edge cases I believe. 
2. The whole point here is to generate the systemd unit first, which svc does not (and should not?) do. I was considering to move the systemd stuff to some common utility code, so that mount and svc could both rely on that. But with a very small patch to svc, it was embeddable as you had proposed originally. This seems like the right approach to me atm.
3. That shell test is basically a placeholder. It does not run. I might go ahead and propose and prototype a better approach some time, and when I do, this test should not be forgotten. So having the placeholder is valuable from my point of view.

File: engine/resources/mount.go [hunk 1]
```diff
@@ -243,36 +323,36 @@ func (obj *MountRes) Watch(ctx context.Context) error {
...
-	recWatcher, err := recwatch.NewRecWatcher(fstabPath, false)
-	if err != nil {
-		return err
-	}
-	// close the recwatcher when we're done
-	defer recWatcher.Close()
-
 	obj.init.Running() // when started, notify engine that we're running
 
+	// // watch the unit file
```
Comment (ffrank) [inline engine/resources/mount.go:328]: This should probably be uncommented, but I want better end-to-end testing facilities first. These wrapped resources make life a little bit complicated.

File: engine/resources/mount.go [hunk 2]
```diff
@@ -300,86 +380,49 @@ func (obj *MountRes) Watch(ctx context.Context) error {
...
-	}
-	obj.init.Logf("mountCheckApply(%t)", apply)
+	obj.init.Logf("finished CheckApply on unit file (return %v)", checkOK)
 
-	if obj.State == "exists" {
-		// Reload mounts from /etc/fstab by performing a `daemon-reload` and
-		// restarting `local-fs.target` and `remote-fs.target` units.
-		if err := mountReload(ctx); err != nil {
-			return false, errwrap.Wrapf(err, "error reloading /etc/fstab")
+	if !checkOK {
```
Comment (ffrank) [inline engine/resources/mount.go:409]: ah shoot this needs an `&& apply`

**Comment by purpleidea** [comment]
FYI as per Felix good idea we're in https://github.com/purpleidea/mgmt/issues/745 for now until we agree on how to proceed.
